Add check for project name

## Version
v2.1.1

## Reproduction link
https://github.com/nuxt/create-nuxt-app

## Steps to reproduce
Create a Nuxt project with `create-nuxt-app <name>`

Then, CLI asks for a project name, description etc… If project name is invalid, CLI doesn't handle it, but project can't be run with npm run dev.

`$ npm run dev
npm ERR! Invalid name: "Hugo Torzuoli"`

## What is expected ?
CLI should fire an error/warning if name is invalid

## What is actually happening?
Nothing during CLI steps, error when serving app.

--- a/package.json
+++ b/package.json
@@ -21,7 +21,8 @@
     "lodash": "^4.17.4",
     "minimist": "^1.2.0",
     "sao": "^0.22.5",
-    "superb": "^2.0.0"
+    "superb": "^2.0.0",
+    "validate-npm-package-name": "^3.0.0"
   },
   "devDependencies": {
     "ava": "^0.23.0"

--- a/sao.js
+++ b/sao.js
@@ -2,6 +2,7 @@ const superb = require('superb')
 const glob = require('glob')
 const { join } = require('path')
 const spawn = require('cross-spawn')
+const validate = require("validate-npm-package-name")
 
 const rootDir = __dirname
 
@@ -120,6 +121,15 @@ module.exports = {
     '.prettierrc': 'prettier === "yes"'
   },
   move(answers) {
+    const validation = validate(answers.name)
+    validation.warnings && validation.warnings.forEach(warn => {
+      console.warn('Warning:', warn)
+    })
+    validation.errors && validation.errors.forEach(err => {
+      console.error('Error:', err)
+    })
+    validation.errors && validation.errors.length && process.exit(1)
+
     const moveable = {
       gitignore: '.gitignore',
       '_package.json': 'package.json',

